### PR TITLE
appveyor: upgrade VS2017 to VS2019

### PR DESCRIPTION
**Pull request by bagder**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -46,9 +46,9 @@ environment:
         SHARED: OFF
         DISABLED_TESTS: ""
         COMPILER_PATH: ""
-      - APPVEYOR_BUILD_WORKER_IMAGE: "Visual Studio 2017"
+      - APPVEYOR_BUILD_WORKER_IMAGE: "Visual Studio 2019"
         BUILD_SYSTEM: CMake
-        PRJ_GEN: "Visual Studio 15 2017 Win64"
+        PRJ_GEN: "Visual Studio 15 2019 Win64"
```
Comment (MarcelRaad) [inline appveyor.yml]: The deprecated architecture selection via the generator name has been removed from the VS 2019 generator, it has to be specified with "-A x64" now: https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2016%202019.html

Comment (bagder) [reply appveyor.yml]: Aha, thanks. I'll amend!

**Comment by bagder** [comment]
Ugha, that failed so spectacularly I'm afraid I don't know where to go next...

**Comment by jay** [comment]
It looks like there's an extra space
 Configuration='Debug'  Platform=' x64'.

**Comment by bagder** [comment]
hm, yes. So where does that originate from? The documentation says `-A x64" so surely that's not the space?

**Comment by bagder** [comment]
Hm, I bet the quotes make it so...

**Comment by jay** [comment]
I updated my comment  after I made it, my guess you are quoting so it sees it as one argument
"-A x64"
then I guess cmake parses it as -A " x64"

**Comment by bagder** [comment]
I'm testing this out in the #4387 PR

**Pull request closed without merging**